ios/LayoutReanimation/REASharedTransitionManager.m

```diff
@@ -0,0 +1,512 @@
+  [self setValue:[self valueForKey:@"activityState"] forKey:@"activityState"];
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath
+                      ofObject:(id)object
+                        change:(NSDictionary<NSKeyValueChangeKey, id> *)change
+                       context:(void *)context
+{
+  UIView *screen = (UIView *)object;
+  UIView *stack = screen;
+  while (stack != nil && ![stack isKindOfClass:[RNSScreenStackView class]]) {
+    stack = stack.superview;
+  }
+
+  if ([keyPath isEqualToString:@"transitionDuration"]) {
+    // added new screen (push)
+    if (screen.superview != nil) {
+      [self runAsyncSharedTransition];
+    }
+  } else if ([keyPath isEqualToString:@"activityState"]) {
+    // removed screen from top (removed from stack or covered by another screen)
+    bool isRemovedInParentStack = [self isRemovedFromHigherStack:screen];
+    if (stack != nil && !isRemovedInParentStack) {
+      if ([self isScreen:screen outsideStack:stack]) {
+        // screen is removed from React tree ( navigation.navigate(<screenName>) )
+        [self runSharedTransitionForSharedViewsOnScreen:screen];
+      } else if ([self isScreen:screen onTopOfStack:stack]) {
+        // click on button goBack on native header
+        [self runSharedTransitionForSharedViewsOnScreen:screen];
+      } else {
+        [_animationManager visitTree:screen
+                               block:^int(id<RCTComponent> view) {
+                                 if ([self->_animationManager hasAnimationForTag:view.reactTag
+                                                                            type:@"sharedElementTransition"]) {
+                                   REASnapshot *snapshot = [[REASnapshot alloc] init:(UIView *)view
+                                                                          withParent:((UIView *)view).superview];
+                                   self->_snapshotRegistry[view.reactTag] = snapshot;
+                                 }
+                                 return false;
+                               }];
+      }
+    } else {
+      // removed stack
+      for (UIView *child in stack.reactSubviews) {
+        [_animationManager visitTree:child
+                               block:^int(id<RCTComponent> _Nonnull view) {
+                                 [self clearAllSharedConfigsForView:(UIView *)view];
+                                 return false;
+                               }];
+        [_screenHasObserver removeObject:child.reactTag];
+      }
+    }
+  }
+}
+
+- (BOOL)isScreen:(UIView *)screen outsideStack:(UIView *)stack
+{
+  for (UIView *child in stack.reactSubviews) {
+    if ([child.reactTag isEqual:screen.reactTag]) {
+      return NO;
+    }
+  }
+  return YES;
+}
+
+- (BOOL)isScreen:(UIView *)screen onTopOfStack:(UIView *)stack
+{
+  int screenCount = stack.reactSubviews.count;
+  return screenCount > 0 && screen == stack.reactSubviews[screenCount - 1];
```

Comment:
I think you can use `reactSubviews.lastObject` instead of `reactSubviews[screenCount - 1]`.